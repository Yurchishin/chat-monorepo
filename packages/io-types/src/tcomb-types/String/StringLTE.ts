import t from 'tcomb-validation'
import { StringLT } from './StringLT'
import { StringEqualLength } from './StringEqualLength'

export const StringLTE = (min: number) => t.union([
    StringLT(min),
    StringEqualLength(min),
], 'StringLTE')
