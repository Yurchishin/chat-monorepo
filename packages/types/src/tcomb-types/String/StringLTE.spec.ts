import {StringLTE} from './StringLTE'
import {expectTypeSuccess, expectTypeFailure} from "../../tests"

describe('StringLTE(5)', () => {
	const LTE5 = StringLTE(5)

	expectTypeSuccess(LTE5, '4444')
	expectTypeSuccess(LTE5, '55555')
	expectTypeFailure(LTE5, '666666')
	expectTypeFailure(LTE5, 1)
})
