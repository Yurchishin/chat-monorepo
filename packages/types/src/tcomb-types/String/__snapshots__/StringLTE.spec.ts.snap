// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringLTE(5) expectTypeFailure(1): TYPE_FAILURE::StringLTE(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Invalid value 1 supplied to StringLTE",
      "path": Array [],
    },
  ],
  "value": 1,
}
`;

exports[`StringLTE(5) expectTypeFailure(666666): TYPE_FAILURE::StringLTE(666666) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "666666",
      "expected": [Function],
      "message": "Invalid value \\"666666\\" supplied to StringLTE",
      "path": Array [],
    },
  ],
  "value": "666666",
}
`;
